src/main/java/io/dropwizard/sharding/dao/LookupDao.java

```diff
@@ -430,6 +428,39 @@ public LockedContext(int shardId, SessionFactory sessionFactory, Function<T, T>
             });
         }
 
+        public<U> LockedContext<T> selectAndUpdateOrSave(RelationalDao<U> relationalDao, DetachedCriteria criteria, Function<U, U> handler) {
+            return apply(parent-> {
+                try {
+                    relationalDao.selectAndUpdateOrSave(this, criteria, handler);
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+                return null;
+            });
+        }
+
+        public<U> LockedContext<T> scrollAndUpdate(RelationalDao<U> relationalDao, DetachedCriteria criteria, ScrollMode scrollMode, Function<ScrollableResults,List<U>> handler) {
+            return apply(parent-> {
+                try {
+                    relationalDao.scrollAndUpdate(this, criteria, scrollMode, handler);
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+                return null;
+            });
+        }
+
+        public<U> LockedContext<T> updateAll(RelationalDao<U> relationalDao, DetachedCriteria criteria, int start, int numRows, Function<List<U>, List<U>> handler) {
```

Comment:
This is redundant. scrolling update method above should be sufficient. We shall never want to update only n objects. Even if it si needed, it can be specified in the criteria itself. Remove this.